cmd/bootstrap-post/main.go

```diff
@@ -1,80 +1,130 @@
+
+	conf, err := config.NewConfig()
 	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
+		exit(log, err)
 	}
 
-	err = readiness.CheckIfRequiredDeploymentsAreReady()
+	level, err := logrus.ParseLevel(conf.Log)
 	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
-	}
+		log.Infof(
+			"Failed to parse log level %v, continuing with the default log level Info, err %v", conf.Log, err,
+		)
 
-	tp := &http.Transport{
-		TLSClientConfig: &tls.Config{InsecureSkipVerify: data.BootstrapConf.InsecureSkipVerify},
+		level = logrus.InfoLevel
 	}
-	client := http.Client{Transport: tp}
 
-	fmt.Println("Started creating dashboard org")
+	log.SetLevel(level)
+	log.WithField("level", level.String()).Debug("Set the log level")
 
-	err = helpers.CheckForExistingOrganisation(client)
+	k8sClient, err := k8s.NewClient(conf, log.WithField("Client", "Kubernetes"))
 	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
+		exit(log, err)
 	}
 
-	fmt.Println("Finished creating dashboard org")
-	fmt.Println("Generating dashboard credentials")
-
-	err = helpers.GenerateDashboardCredentials(client)
-	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
+	if err = k8sClient.CheckIfRequiredDeploymentsAreReady(); err != nil {
+		exit(log, err)
 	}
 
-	fmt.Println("Finished generating dashboard credentials")
-	fmt.Println("Started bootstrapping operator secret")
+	orgExists := false
 
-	if data.BootstrapConf.OperatorKubernetesSecretName != "" {
-		err = helpers.BootstrapTykOperatorSecret()
-		if err != nil {
-			fmt.Println(err)
-			os.Exit(1)
+	tykSvc := tyk.NewService(conf, log.WithField("Client", "Tyk"))
+	if err = tykSvc.OrgExists(); err != nil {
+		if !errors.Is(err, tyk.ErrOrgExists) {
+			exit(log, err)
 		}
+
+		orgExists = true
 	}
 
-	fmt.Println("Finished bootstrapping operator secret\nStarted bootstrapping portal secret")
+	if !orgExists {
+		if conf.BootstrapDashboard {
+			log.Info("Bootstrapping Tyk Dashboard")
+
+			if err = tykSvc.CreateOrganisation(); err != nil {
+				exit(log, err)
+			}
+
+			if err = tykSvc.CreateAdmin(); err != nil {
+				exit(log, err)
+			}
+		}
+
+		if conf.BootstrapPortal {
+			log.Info("Bootstrapping Tyk Classic Portal")
+
+			if err = tykSvc.BootstrapClassicPortal(); err != nil {
+				exit(log, err)
+			}
 
-	if data.BootstrapConf.DevPortalKubernetesSecretName != "" {
-		err = helpers.BootstrapTykPortalSecret()
-		if err != nil {
-			fmt.Println(err)
-			os.Exit(1)
+			if err = k8sClient.RestartDashboard(); err != nil {
+				exit(log, err)
+			}
 		}
 	}
 
-	fmt.Println("Started bootstrapping portal with requests to dashboard")
+	// Common log message for organization existence
+	if orgExists && conf.DevPortalKubernetesSecretName != "" || conf.OperatorKubernetesSecretName != "" {
+		log.WithFields(logrus.Fields{
+			"organisationName":  conf.Tyk.Org.Name,
+			"organisationCName": conf.Tyk.Org.Cname,
+		}).Info("Organisation exists on Tyk. " +
+			"Please provide the Organisation ID and Dashboard Access Key for Kubernetes secrets")
```

Comment:
Actually, this log's intent is to remind users to provide credentials while the organisation exists and the user wants to create Kubernetes secrets. Can you please elaborate why we need to change the log as suggested?

Reply:
This log will be printed during bootstrapping right?
They should have it already created before bootstrapping.

Reply:
Yes, it is printed during bootstrapping. I just added this print statement to remind users if they have bootstrapped Tyk Dasboard, they need to provide OrgID and Authentication key via environment variables. let's assume I have bootstrapped Tyk Dashboard, in that case, we need to get organisation id and user's access_key as input from the user since these values are used in secrets.